plugins/put-a-file-with-ftp/impl/src/main/java/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUT.java

```diff
@@ -639,8 +642,16 @@ public Result execute( Result previousResult, int nr ) {
         }
       }
 
+      Path path = Paths.get( realLocalDirectory );
```

Comment:
You need to add unit tests for this new edge cases for the logic added in JobEntryFTPPUT.java#execute
There is an existing class:
https://github.com/pentaho/pentaho-kettle/blob/master/plugins/put-a-file-with-ftp/impl/src/test/java/org/pentaho/di/job/entries/ftpput/JobEntryFTPPUTTest.java

I understand the function is 200+ lines with many many edge cases. I'm only asking the two exceptions you just added.

Reply:
fixed with https://github.com/pentaho/pentaho-kettle/commit/42a31cf00e0008515d6a51939fcac7cd624e1075